src/api/forms/service.js

```diff
@@ -1,25 +1,140 @@
+  const formConfiguration = { ...formConfigurationInput, id: formId }
+
+  // create the form object. At this point, we're just creating a blank
+  // form following the one-page template, we just set the title per the user request.
+  const shallowCloneForm = { ...emptyForm, name: formConfiguration.title }
+
+  const validationResult = Schema.validate(shallowCloneForm)
+  if (validationResult.error) {
+    throw new InvalidFormDefinitionError(validationResult.error.toString())
+  }
+
+  try {
+    await createFormDefinition(formConfiguration, shallowCloneForm)
+    await createFormMetadata(formConfiguration)
+  } catch (error) {
+    logger.error(error, "Failed to persist, couldn't create form.")
+    throw new FailedCreationOperationError()
+  }
+
+  return formConfiguration
+}
+
 /**
  * Lists the available forms
  * @returns {Promise<FormConfiguration[]>} - form configuration
  */
-export function listForms() {
+export async function listForms() {
   return listFormMetadataEntries()
 }
 
 /**
- * Retrieves a form (metadata + form definition)
+ * Retrieves a form configuration
  * @param {string} formId - ID of the form
  * @returns {Promise<FormConfiguration>} - form configuration
  */
-export function getForm(formId) {
+export async function getForm(formId) {
   return getFormMetadata(formId)
 }
 
+/**
+ * Retrieves the form definition for a given form ID
+ * @param {string} formId - the ID of the form
+ * @returns {Promise<string>} - form definition JSON content
+ */
+export async function getFormDefinition(formId) {
+  return getFormDefinitionFromRepository(formId)
+}
+
+/**
+ * Given a form title, returns the ID of the form.
+ * E.g. "Hello - world" -> "hello-world".
+ * @param {string} formTitle - title of the form
+ * @returns {string} - ID of the form
+ */
+function formTitleToId(formTitle) {
+  return formTitle
+    .toLowerCase()
+    .replace(/[^a-z0-9 ]/g, '') // remove any non-alphanumeric characters
+    .replace(/\s+/g, ' ') // replace any whitespaces with a single space
+    .replace(/ /g, '-') // replace any spaces with a hyphen
+}
+
+/**
+ * Retrieves the empty form configuration
+ * @returns {Promise<object>} - the empty form configuration
+ * @throws {InvalidFormDefinitionError} - if the base form definition is invalid
+ */
+async function retrieveEmptyForm() {
+  const fileContent = await readFile(
+    new URL('./empty-form.json', import.meta.url).pathname,
+    'utf-8'
+  )
+
+  try {
+    const emptyForm = JSON.parse(fileContent)
+
+    const validationResult = Schema.validate(emptyForm)
+
+    if (validationResult.error) {
+      throw new InvalidFormDefinitionError(
+        'Invalid form schema provided. Please check the empty-form.json file.'
+      )
+    }
+
+    return emptyForm
+  } catch (error) {
+    if (error instanceof SyntaxError) {
+      throw new InvalidFormDefinitionError(
+        'Failed to parse empty-form.json as JSON. Please validate contents.'
+      )
+    }
+
+    throw error
+  }
```

Comment:
Another one where we can keep the message more generic, but add the real cause to stack traces?

```suggestion
  } catch (cause) {
    throw new InvalidFormDefinitionError(
      'Failed to parse empty-form.json as JSON. Please validate contents.',
      { cause }
    )
  }
```